pymc3/model.py

```diff
@@ -1695,7 +1695,13 @@ def pandas_to_array(data):
     XXX: When `data` is a generator, this will return a Theano tensor!
 
     """
-    if hasattr(data, "values"):  # pandas
+    if isinstance(data, np.ndarray):
```

Comment:
This check evaluates True to masked arrays, meaning you will never get to the `elif hasattr(data, "mask")` below.

```python
a = np.ma.array(np.ones(3), mask=[0, 0, 1])  
isinstance(a, np.ndarray)  # Returns True                                        
```

That check's only purpose seems to be to fill empty masks, so we should do that here. And also maybe not mask already created masked arrays that still include nans? Assuming there might be other reasons to leave nans out of a np.ma.array?

Reply:
Yes, good catch. I re-worked the branching logic.